### DISCUSSION TITLE
Wiki-link in HTML?

### DISCUSSION THREAD

**Discussion by ouuan** [srid/neuron#641] [Ideas]
It seems that wiki-links don't work inside HTML. e.g. `[[id]]` works but `<p>[[id]]</p>` doesn't work.

Is there/Will there be a solution for this?

**Comment by ouuan** [comment]
Currently, I'm using a very hacky and ugly JavaScript workaround for this.

**Comment by srid** [comment]
Markdown within HTML is not recognized by the commonmark parser. You can try asking in [commonmark-hs](https://github.com/jgm/commonmark-hs) tracker (neuron uses that library).

**Reply by ouuan** [reply]
Thanks for pointing this out! I searched for "markdown in HTML" and found [How can I wrap my markdown in an HTML div? - Stack Overflow](https://stackoverflow.com/questions/29368902/how-can-i-wrap-my-markdown-in-an-html-div). Now I'm able to solve this by adding two blank lines like [this](https://babelmark.github.io/?text=%3Cdl%3E%0A%3Cdt%3E%0A**foo**%0A%3C%2Fdt%3E%0A%3Cdd%3Ebar%3C%2Fdd%3E%0A%3C%2Fdl%3E%0A%0A%3Cdl%3E%0A%3Cdt%3E%0A%0A**foo**%0A%0A%3C%2Fdt%3E%0A%3Cdd%3Ebar%3C%2Fdd%3E%0A%3C%2Fdl%3E).

**Reply by srid** [reply]
Note that you can also use commonmark's attributes extension (enabled in neuron) to wrap parts of your markdown in a custom-classed div or span, thus obviating raw HTML while allowing certain use cases if not all.

[Emanote supports it too](https://note.ema.srid.ca/demo/markdown) (see the 'Note' box at the top). Another example, the first paragraph of https://ema.srid.ca/ (appears bigger) or the "Next" link styling in https://ema.srid.ca/start etc

**Reply by srid** [reply]
Specifically https://github.com/jgm/commonmark-hs/blob/master/commonmark-extensions/test/fenced_divs.md
